### PR TITLE
Move some tests to integration

### PR DESCRIPTION
**Pull request by Lorak-mmk**
This is the first step towards improved integration tests.
Tests that require a running cluster should be in integration tests.

This PR moves some of such tests. Not all - for that there are a few other changes needed, but I wanted to merge what I already have.

Ref: https://github.com/scylladb/scylla-rust-driver/issues/1097
Fixes: https://github.com/scylladb/scylla-rust-driver/issues/1048

## Pre-review checklist

- [x] I have split my patch into logically separate commits.
- [x] All commit messages clearly explain what they change and why.
- [ ] I added relevant tests for new features and bug fixes.
- [x] All commits compile, pass static checks and pass test.
- [x] PR description sums up the changes and reasons why they should be introduced.
- [ ] I have provided docstrings for the public items that I want to introduce.
- [ ] I have adjusted the documentation in `./docs/source/`.
- [ ] I added appropriate `Fixes:` annotations to PR description.

### CODE REVIEW

File: scylla/tests/integration/utils.rs
Comment (wprzytula) [inline scylla/tests/integration/utils.rs:36]: The counter is not shared between `scylla` crate and the integration tests, yet it's a separate counter. Won't there be a name slash between unit tests still using Scylla and integration tests?
Maybe a better solution would be to have this counter exposed in `test_utils` in `scylla` and shared with integration tests?

Comment (Lorak-mmk) [reply scylla/tests/integration/utils.rs:36]: Moving to test_utils in scylla doesn't address this issue at all. Unit tests and integration tests are still 2 separate crates, and don't share memory - so the counters will be separate.

This is however not an issue: those binaries are run sequentially afaik, and current timestamp is included in the name, so there will be no clash.

In the future, the only counter will be in integration tests because unit tests won't connect to database.

**Comment by Lorak-mmk** [comment]
1st push: rebased on main
2nd push: Fixed errors, fixed minor related issue (https://github.com/scylladb/scylla-rust-driver/issues/1048)

**Comment by Lorak-mmk** [comment]
Fixed errors in serverless tests

File: scylla/src/lib.rs
Comment (muzarski) [inline scylla/src/lib.rs:250]: The docstring is outdated - it mentions that module is `pub`.

**Review by muzarski — APPROVED** [review]

File: scylla/src/lib.rs
Comment (Lorak-mmk) [reply scylla/src/lib.rs]: I removed the comment.

**Review by wprzytula — DISMISSED** [review]
Please don't merge it before #1057 is merged - it's going to introduce a lot of painful conflicts for me.

**Review by muzarski — APPROVED** [review]

**Pull request merged**
